### PR TITLE
[Testing:TAGrading] Testing TA Grading hotkeys

### LINKED ISSUE #10501: Cypress test for TA Grading Keyboard shortcuts
Writing the e2e test for this hotkey as i mentioned below and also attached the image for it. If we break the code in it the cypress code can help us and also reduce the time in debugging

**Describe the way you'd like to solve this problem**

Here is a list of shortcuts so far, (as per this pr #1995 )
* A: autograding module
* G: grading rubric module
* O: file browser module
* S: student info module
* 1 - 9: select/deselect rubric mark (of currently open component)

Screenshot 2024-06-02 at 6 29 40 PM

### PR DESCRIPTION
**Pull request by manzil-infinity180**
Closes #10501 
Testing the interface of ta grading on keyboard events 
Here is the list of hotkeys(keyboard shortcut)

Screenshot 2024-06-02 at 6 29 40 PM

### CODE REVIEW

**Review by zacharymnp — APPROVED** [review]
good tests; could expand them some

File: site/app/templates/grading/electronic/NotebookPanel.twig
Comment (IDzyre) [inline site/app/templates/grading/electronic/NotebookPanel.twig:6]: Looks like a merging issue...

**Comment by IDzyre** [comment]
@manzil-infinity180 The tests don't work locally on my machine, i believe due to grade inquiries not being enabled

**Comment by manzil-infinity180** [comment]
> @manzil-infinity180 The tests don't work locally on my machine, or in CI. I am not sure why, but the data-testid for the add peer component doesn't attach to the button.

I have tested my code it works until notebook has added
When i add the notebook from notebook builder (sample/gradeable/grading_only_peer_team_homework/update?nav_tab=1) some syntax error for `showPopup` and `colorEvent` as you can see in image
Screenshot 2024-06-08 at 3 47 07 PM

**Comment by manzil-infinity180** [comment]
@IDzyre  Look Test passed for hotkey , now failing for some other cypress test 

Just having some issue with notebook builder from my side

File: site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js
Comment (IDzyre) [inline site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js:33]: The reason the other test is failing is because it is assuming that there is no peer component to grade for this gradeable. As an after() block could you remove the peer component? (Or create a before() statement in the test_rubric_access test to remove all peer components

As a side note, it would be good to test that adding a peer component allows students to peer grade.

**Review by IDzyre — CHANGES_REQUESTED** [review]

File: site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js
Comment (manzil-infinity180) [reply site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js:41]: yeah it works , 
Ok i will do it for peer grade component

File: site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js
Comment (IDzyre) [inline site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js:46]: Is this still being used? if so, uncomment it so we can see if the tests pass, if not, remove it.

File: site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js
Comment (IDzyre) [inline site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js:55]: Same about comments

File: site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js
Comment (IDzyre) [inline site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js:5]: I don't think this is the best way to do this, while this may do it correctly in an ideal world, it will still leave peer components if the test is re-run multiple times without hitting the after block. I am going to tag @williamjallen to see if he thinks this is good enough, (as it will work in CI, but may cause problems to local users). 

At the very least, you can test that after you click the 'delete gradeable component' button, assert that there are no more peer-components left.

**Review by IDzyre — CHANGES_REQUESTED** [review]

File: site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js
Comment (IDzyre) [inline site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js:16]: For the testing of the hotkeys, on my system it is flaky if you use the nav-bar as the place you are wanting to type. After some research, many people just do cy.get('body').type({letter}); to use hotkeys, and after testing with my system, it works more consistently. Can you test this on your machine, and maybe a couple of other people to see if it is just my machine ?

File: site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js
Comment (manzil-infinity180) [reply site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js:5]: What I think why the CI passed , as there is no peer component and then i add one peer component and later i deleted that one , so i am adding one line for assertion that no more peer component left 

later i will change if i get any other way to do this

File: site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js
Comment (manzil-infinity180) [reply site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js:16]: yeah its working fine

File: site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js
Comment (IDzyre) [reply site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js:5]: After discussing with William, I think this is fine for now, just add the assertion that the peer component doesn't exist, and I'll create an issue to refactor some of the older tests that require the state of the VM to be in a certain way, to use before blocks to set the state.

File: site/app/templates/admin/admin_gradeable/AdminGradeableAuto.twig
Comment (IDzyre) [inline site/app/templates/admin/admin_gradeable/AdminGradeableAuto.twig:104]: Is this change necessary? If not it'd be better to remove it. After this it looks good to merge.

**Review by IDzyre — CHANGES_REQUESTED** [review]
One small question about a spacing change then looks good to merge.

File: site/public/js/notebook_builder/widgets/widget.js
Comment (IDzyre) [inline site/public/js/notebook_builder/widgets/widget.js:87]: Another spacing change that isn't needed

**Review by IDzyre — CHANGES_REQUESTED** [review]
Last spacing change then i'll merge if no more are added

File: site/app/templates/grading/electronic/GradingPanelHeader.twig
Comment (IDzyre) [inline site/app/templates/grading/electronic/GradingPanelHeader.twig:1]: Are you adding any testids to this file ? If not can you remove these spacing changes

**Review by IDzyre — CHANGES_REQUESTED** [review]

**Review by IDzyre — APPROVED** [review]
Good work, looks good.

**Pull request merged**
